dex/elrond_dex_farm/src/lib.rs

```diff
@@ -476,13 +495,13 @@ pub trait Farm:
             &new_farm_contribution,
             &farm_token_id,
             &new_attributes,
-            &caller,
+            &payments[1..],
```

Comment:
why do you start from 1 ?

Reply:
Isn't it better to refactor all this? Like making a different merge function? Which gets all the payments, merges them, burns them and creates the new one?